crates/shared/src/price_estimation/competition.rs

```diff
@@ -409,6 +438,7 @@ mod tests {
 
     #[tokio::test]
     async fn works() {
+        tracing_subscriber::fmt::init();
```

Comment:
This should either use `observe::tracing::initialize_reentrant("some_filter")` or needs to be dropped. This is what caused the test to fail although the logic works.